File: 프로그래머스/lv1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        // 방법1
        s = s.toLowerCase();
        int pCnt = s.length() - (s.replace("p","")).length();
        int yCnt = s.length() - (s.replace("y","")).length();
        
        if(pCnt != yCnt) answer = false;
        
        // 방법2
        /*s = s.toUpperCase();
        int pCnt = 0;
        int yCnt = 0;
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i) == 'Y'){
                yCnt++;
            } else if(s.charAt(i) == 'P'){
                pCnt++;
            }
        }   
        if(yCnt != pCnt) answer = false;*/
        
        return answer;
    }
}

File: 프로그래머스/lv1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.js
function solution(s){
    var answer = true;
    
    let pCnt = s.length - s.replace(/p/gi,'').length;
    let yCnt = s.length - s.replace(/y/gi,'').length;
    
    answer = (pCnt!=yCnt) ? false : true;

    return answer;
}